src/Moq/MethodCallReturn.cs

```diff
@@ -171,6 +171,13 @@ private void ValidateReturnDelegate(Delegate callback)
 		{
 			var callbackMethod = callback.GetMethodInfo();
 
+			var expectedParams = this.Method.GetParameters();
+			var actualParams = callback.GetMethodInfo().GetParameters();
+			if (callback.HasCompatibleParameterList(expectedParams))
+			{
+				return;
```

Comment:
I still don't understand why we are making `Returns` more permissive by possibly returning early and skipping sensible sanity checks here. *Why* are you proposing this change? Which specific use case would profit from it? Have you seen the 3 previous issues/PRs I linked to, over in the issue?

Reply:
(Btw., it would be ideal to document the use cases that profit from the proposed change in the form of one or more tests.)

Reply:
1. I'm going to be more permissive exactly in places where it's OK to be, i.e. when the method is runnable. The HasCompatibleParameterList method returns true only if it can be DynamicInvoked or Invoked, so when it returns true we are sure that it's OK. I've seen the old three issues and I think that they all points to the direction I'm going to.
2. The sanity checks were proved as wrong. We had one case (in https://github.com/moq/moq4/issues/652) for which it wrongly failed. The fix in the previous PR targets exactly the problem that was caught, but we don't know if there are other cases that .NET manipulates the method signature like the Closure argument in https://github.com/moq/moq4/issues/652. IMO Checking the standard .NET Invoke method is a more stable way to eliminate such manipulations than implementing a heuristics and extend it on each case we encounter (as was really proved in the Closure bug).
3. That's also the answer for the use case question - the use case is https://github.com/moq/moq4/issues/652 . Although was already targeted by another way in a previous PR, it may be only one example among others (still unknown) cases in which the existing heuristics doesn't cover.